functions/patch3API.js

```diff
@@ -0,0 +1,39 @@
+const functions = require("firebase-functions");
+const cors = require("cors")({ origin: true });
+
+const admin = require("firebase-admin");
+
+const db = admin.firestore();
+
+exports.patch3 = functions.https.onRequest((request, response) =>{
+    cors(request, response, () =>
+    {
+        const OrgDomain = request.body.data.OrgDomain;
+        const oldField = request.body.data.FieldName;
+        const oldFieldValue = request.body.data.FieldValue;
+        const newField = request.body.data.NewField;
+        const newFieldValue = request.body.data.NewFieldValue;
+        let taskId;
+        
+        const promise1 = db.collection("Organizations").doc(OrgDomain).collection("Tasks").where(oldField, "==", oldFieldValue).get().then((task) => {
+            task.forEach((doc) => {
+                taskId = doc.data().Id;
+                console.log("Executing Promise2 of Patch3");
+        desiredField = {};
+        desiredField[newField]=newFieldValue;
```

Comment:
This is normally a key value, json works in that way. So please check if we can improve this representation.